### PR TITLE
Adeopura 021616 fixes

### PR DESCRIPTION
**Pull request by adeopura3**
Removed unnecessary files
Changes to enqueue and dequeue algorithm
Removed promises from the functions
Update the template and the test files

### CODE REVIEW

File: app/directives.js [hunk 1]
```diff
@@ -4,71 +4,12 @@ airTrafficControlApp.directive('aircraftDetails', function(){
       templateUrl:'directives/aircraftDetails.html',
       scope:{
          aircraftObject: "="
```
Comment (moralesalberto) [inline app/directives.js]: Please remove this; accomplish this in a different way; this is a 'weird' implementation, and we do not want these.

File: app/services.js [hunk 2]
```diff
@@ -19,29 +19,7 @@ airTrafficControlApp.service('systemBootService', ['$location', '$q', '$timeout'
...
-                // Resolve the deferred to indicate
-                // success
-                deferred.resolve(self.systemBootStatus);
-                
-            }, 50);
-        }
-        
-        // Make sure to return a promise from the async call
-        return deferred.promise;
+        self.systemBootStatus = 'Booted';
```
Comment (moralesalberto) [inline app/services.js]: what is bootSystem doing? I would have thought it clears the queue, but here it simply just sets a string?

File: app/services.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // Services
 
 // Boot service
-airTrafficControlApp.service('systemBootService', ['$location', '$q', '$timeout', function($location, $q, $timeout){
+airTrafficControlApp.service('systemBootService', ['$location', function($location){
```
Comment (moralesalberto) [inline app/services.js:6]: Please use the angular factory method to create your service; again do things the angular way, the way the tutorial shows you how to create a service.
Also use the module pattern, return an object with the functions you want available.

File: app/services.js [hunk 4]
```diff
@@ -53,229 +31,99 @@ airTrafficControlApp.service('systemBootService', ['$location', '$q', '$timeout'
...
+        "aicraftQueue": []
+    },
+    {
+        "type": "Cargo",
+        "size": "Small",
+        "aicraftQueue": []
     }
+    ];
       
     this.initialize = function(){
```
Comment (moralesalberto) [inline app/services.js]: Is this systemBoot? at least it needs to be called from systemBoot()

File: app/services.js [hunk 5]
```diff
@@ -53,229 +31,99 @@ airTrafficControlApp.service('systemBootService', ['$location', '$q', '$timeout'
...
-
-}]);
-
-
 
 
 // Queue management service [Enqueue/Dequeue]
 // Use the array to avoid the minification problem
-airTrafficControlApp.service('queueManagementService', ['$q', '$timeout', 'aircraftPriorityDeterminationService', function($q, $timeout, aircraftPriorityDeterminationService) {
+airTrafficControlApp.service('queueManagementService', [function() {
```
Comment (moralesalberto) [inline app/services.js]: 1) use module pattern
2) please add comments to the top of this function explaining the functionality of this service. It seems overly complicated; the contract per the assignment is that it responds to: boot, enqueue(aircraft) and dequeue();

so three public methods, and possibly a few private ones to handle this; but I am reading a lot more functions for those three simple actions.

File: app/services.js (hunk 2 above)
Comment (adeopura3) [inline app/services.js]: The boot system was just storing the boot status. The queueManagementService was being initialized in the controller once the user clicks boot system. I have changed the code to initialize the queue in the bootSystem function. That makes sense

File: app/services.js (hunk 4 above)
Comment (adeopura3) [inline app/services.js]: Added this to system boot

File: app/services.js (hunk 5 above)
Comment (adeopura3) [inline app/services.js]: I have 5 public functions in the queueManagementService.

initialize: For initialization

numberOfAircraftsInQueue: To enable / disable the "Dequeue Aircraft" button. This is not per the specs and can be removed but I added it to make the app more complete

getAllAircraftsInSystemInPriorityOrder: To Display the list of aircrafts in sorted order in a table. This is part of the specs (There should be a tab or link to take the user to another view that shows the current state of the enqueued airplanes.)

enqueueItem

dequeueItemWithHighestPriority

File: app/services.js (hunk 4 above)
Comment (adeopura3) [inline app/services.js]: Fixed this

**Pull request merged**